### DISCUSSION TITLE
Add support for nested package.json scripts

### DISCUSSION THREAD

**Discussion by kyle-villeneuve** [npm/feedback#362] [CLI] [answered]
It would be very convenient to support nested package.json scripts to help with organization by environment or other criteria e.g.

```json
"scripts": {
  "development": {
    "start": "..."
   },
   "production": {
     "start": "..."
   }
}
```
```
npm run development:start
```
or if it defaulted to using the value of `NODE_ENV`

**Comment by ljharb** [comment]
How would that work if someone defines both a nested `development` + `start` script, and also a `development:start` script?

**Comment by MylesBorins** [comment, marked as answer]
hey @kyle-villeneuve. If you would like to pursue proposing this feature to the CLI team the next step would be to open an issue over at https://github.com/npm/rfcs to kick off the discussion
